src/helpers/tests/generator.test.ts

```diff
@@ -121,3 +143,253 @@ describe("generateTypes", () => {
+  });
+});
+
+describe("convertFuncSignatureToType", () => {
+  test("converts query methods", () => {
+    const result = generator.convertFuncSignatureToType(
+      "(param1: string, param2: number) => void",
+      "query",
+      "User"
+    );
+    expect(result).toBe("(this: UserQuery, param1: string, param2: number) => UserQuery");
+  });
+
+  test("converts instance methods", () => {
+    const result = generator.convertFuncSignatureToType(
+      "(name: string) => boolean",
+      "methods",
+      "User"
+    );
+    expect(result).toBe("(this: UserDocument, name: string) => boolean");
+  });
+
+  test("converts static methods", () => {
+    const result = generator.convertFuncSignatureToType("() => Promise<void>", "statics", "User");
+    expect(result).toBe("(this: UserModel) => Promise<void>");
+  });
+
+  test("handles missing return type", () => {
+    const result = generator.convertFuncSignatureToType("(param: string) =>", "methods", "User");
+    expect(result).toBe("(this: UserDocument) => any");
+  });
+
+  test("handles empty parameters", () => {
+    const result = generator.convertFuncSignatureToType("() => void", "query", "User");
+    expect(result).toBe("(this: UserQuery) => UserQuery");
+  });
+
+  test("handles invalid function signature", () => {
+    const result = generator.convertFuncSignatureToType("invalid signature", "methods", "User");
+    expect(result).toBe(`(this: UserDocument) => any`);
+  });
+
+  test("handles existing this parameter", () => {
+    const result = generator.convertFuncSignatureToType(
+      "(this: any, name: string) => boolean",
+      "methods",
+      "User"
+    );
+    expect(result).toBe("(this: UserDocument, name: string) => boolean");
+  });
+
+  test("handles complex return types", () => {
+    const result = generator.convertFuncSignatureToType(
+      "(id: string) => Promise<Record<string, any>>",
+      "statics",
+      "User"
+    );
+    expect(result).toBe("(this: UserModel, id: string) => Promise<Record<string, any>>");
+  });
+
+  test("sanitizes model name in type", () => {
+    const result = generator.convertFuncSignatureToType(
+      "(name: string) => boolean",
+      "methods",
+      "user.profile"
+    );
+    expect(result).toBe("(this: UserProfileDocument, name: string) => boolean");
+  });
+
+  test("handles error cases gracefully", () => {
+    const result = generator.convertFuncSignatureToType(
+      // @ts-expect-error - Testing invalid input
+      null,
+      "methods",
+      "User"
+    );
+    expect(result).toBe("(this: UserDocument) => any");
+  });
+});
+
+describe("parseFunctions", () => {
+  test("parses method functions", () => {
+    const funcs = {
+      validatePassword: () => {},
+      updateProfile: () => {},
+      initializeTimestamps: () => {}
+    };
+    const result = generator.parseFunctions(funcs, "User", "methods");
+    expect(result).toBe(
+      "validatePassword: (this: UserDocument, ...args: any[]) => any;\n" +
```

Comment:
If we're getting the default cases here it means there's something wrong. I think the `funcs` test variable you're using needs to have return types and `this` params to get picked up properly.

now I know this logic is already somewhat tested in other tests, so we could mock it completely too, but im not too picky here

Reply:
Now that I understand the codebase a bit better, this is not really the right place/way to test this logic anyways since we don't have access to the document type to tag "this" with. I've deleted these tests for now but nice catch!